contributed_definitions/nyaml/NXmpes_liquid.yaml

```diff
@@ -0,0 +1,197 @@
+        exists: recommended
+        unit: NX_VOLTAGE
+        doc: "Electric field applied to the sample with respect to ground / the system, e.g., via biasing."
+      thickness(NX_FLOAT):
+        exists: recommended
+        unit: NX_LENGTH
+        doc: "Thickness of the sample in normal direction to the surface that is probed, e.g., diameter of capillary (round jets) or thickness of the leaf (flat jets)."
+      flow_rate(NX_FLOAT):
+        exists: recommended
+        unit: NX_VOLUME / NX_TIME
+        doc: "Flow rate of the liquid into the chamber, i.e., throught the microjet capillary (or in case of a flat jet: both capillaries)."
+      shape(NX_CHAR):
+        enumeration: ["round", "flat"]
+      nozzle_material(NX_CHAR):
+        exists: recommended
+        doc: "Material the microjet nozzle is made of (users are encouraged to expand on that in group preparation_description)."
+      depends_on:
+        exists: recommended
+        doc: Connects this liquidjet to its transformation chain.
+      (NXtransformations):
+        exists: recommended
+        doc: "Transformations describing the transformation chain of the liquid jet flow direction and optionally the flat jet normal. The vector of the major flow direction of the liquid jet from the injection into the chamber. Along the cylinder axis for single, cylidrical jets, and parallel to the leaf plane for flatjets."
+        rot_x(NX_NUMBER):
+          \@depends_on(NX_CHAR):
+            doc: "Is '.', since beam axis is assumed to be horizontal (perpendicular to gravity)."
+          \@transformation_type(NX_CHAR):
+            enumeration: ["rotation"]
+          unit: NX_ANGLE
+          \@vector(NX_NUMBER):
+        rot_y(NX_NUMBER):
+          \@depends_on(NX_CHAR):
+            doc: "Link to /NXentry/NXsample/rot_x"
+          \@transformation_type(NX_CHAR):
+            enumeration: ["rotation"]
+          unit: NX_ANGLE
+          \@vector(NX_NUMBER):
+        rot_z(NX_NUMBER):
+          \@depends_on(NX_CHAR):
+            doc: "Link to /NXentry/NXsample/rot_y."
+          \@transformation_type(NX_CHAR):
+            enumeration: ["rotation"]
+          unit: NX_ANGLE
+          \@vector(NX_NUMBER):
+        leaf_normal(NXtransformations):
+          exists: optional
+          doc: "Direction of surface normal vector of the first leaf on the side of solvent #1."
+          rotation_theta(NX_NUMBER):
+            doc: "Direction of surface normal vector of the first leaf on the side of solvent #1. Defaults to be parallel to detection direction."
+            \@depends_on(NX_CHAR):
+              doc: "Link to /NXentry/NXinstrument/NXelectronanalyser/NXcollectioncolumn/NXtransformations/rot_z."
+            \@transformation_type(NX_CHAR):
+              enumeration: ["rotation"]
+            unit: NX_ANGLE
+            \@vector(NX_NUMBER):
+              doc: "Base vector is the liquid jet flow direction."
+      SOLVENT(NXsample_component):
```

Comment:
This part together with the `SOLVENT` is ambiguous. Upper case notation means that the actual entry name can be freely chosen but the group has to have an NX_class attribute, which is `NXsample_component` in this case. However, for both `SOLVENT` and `SOLUTE` this looks the same as they are both an NXsample_component and a verification would not be able to tell whether it should find a `SOLVENT` or `SOLUTE` under this entry.

A simple solution could be to write this as `solvent_SOLVENT` and `solute_SOLUTE`, which means that only the names `solvent_my_freely_chosen_name` and `solute_my_freely_chosen_name` are allowed and everything with a `solvent_` or `solute_` would automatically be required to be of this certain type. @stthuermer Do you think this is a feasible solution here?

Reply:
Yes, this could be a viable solution. Note that the solute is already discernible since it has to include a 'solvent' entry which points to the solvent by containing the SOLVENT name. SOLVENT entries do not have this. I was thinking maybe it is even useful to specify the full path to the SOLVENT inside the 'solvent' entry of each solute instead of just the name, similar to depends_on entries.

Reply:
This is certainly a way to do it. I think it's fine to just use the name of the solvent, for NXtransformations you can also just use the name instead of the full path if they are part of the same NXtransformations group.

For me the question here is whether this is a good approach when we think about general verifiability, because you may know that this field only exists there but people have to implicitly understand this from the appdef and I think it's better to state it explicitly. Also a general algorithm for verification may be a lot simpler, because without prefix it has to build a set of alternatives and test them if any of them fit as a general verification wouldn't know the implicit solvent rule (but I think we have to build this algorithm anyways as this is generally possible with NeXus and then a general verification should also be able to resolve this....).

Reply:
Yes, I very much see the benefit for parsing the file. So lets make it a prefix then. I would actually go for 'solvent' and 'solute' prefixes (i.e., solventSOLVENT and soluteSOLUTE without '_'), because then the groups could then also be named just 'solvent' and 'solute' without any additional characters for just one of each in the entry (the most common case I would say). This is actually how I have currently implemented my exporter: Use 'solvent' / 'solute' for single entries and 'solvent1', 'solvent2' ... and 'solute1', 'solute2' ... for multiple entries. Users can opt to include a underscore character if they want.

Reply:
Sounds good. I added it like this to the file.